tools/merge_archives.ml

```diff
@@ -0,0 +1,106 @@
+(**************************************************************************
+ *                                                                        *
+ *                                 OCaml                                  *
+ *                                                                        *
+ *                    Mark Shinwell, Jane Street Europe                   *
+ *                                                                        *
+ *   Copyright 2021 Jane Street Group LLC                                 *
+ *                                                                        *
+ *   All rights reserved.  This file is distributed under the terms of    *
+ *   the GNU Lesser General Public License version 2.1, with the          *
+ *   special exception on linking described in the file LICENSE.          *
+ *                                                                        *
+ **************************************************************************)
+
+module String_set = Set.Make (String)
+
+let merge_cma ~target:_ ~archives:_ =
+  failwith "not yet implemented"
+
+let read_cmxa filename =
+  let chan = open_in_bin filename in
+  let magic =
+    really_input_string chan (String.length Config.cmxa_magic_number)
+  in
+  let (cmxa : Cmx_format.library_infos) = input_value chan in
+  close_in chan;
+  magic, cmxa
+
+let merge_cmxa0 ~archives =
+  let magic_and_cmxa_list = List.map read_cmxa archives in
+  let magics = List.map fst magic_and_cmxa_list in
+  let cmxa_list = List.map snd magic_and_cmxa_list in
+  let magic =
+    match String_set.elements (String_set.of_list magics) with
```

Comment:
(If the check is added in the function above, this one can
go, and indeed `read_cmxa` does not need to return the
couple.)

Reply:
Now not applicable I think

Reply:
I concur.